src/Perl6/Actions.nqp

```diff
@@ -9185,16 +9187,14 @@ Did you mean a call like '"
                 ),
 
                 # The result of this operation.
-                $<sym> eq 's'
-                ?? QAST::Var.new( :name('$/'), :scope('lexical') )
-                !! QAST::Var.new( :name($S_result), :scope('local') ),
+                $Is_S
+                ?? QAST::Var.new( :name($S_result), :scope('local') )
```

Comment:
Is this reversal of the condition intentional?

Reply:
Yes, as the original test was for 's', not 'S'